docs/DockerContainerLifecycle.md

```diff
@@ -0,0 +1,203 @@
+
+-  Three processes are started:  Felix, confd and Bird (see 
+   [calico/node container components](Components.md) for more details on 
+   these).
+-  Felix is idle, waiting for endpoint data to be configured in etcd.
+-  confd is monitoring the BGP data in etcd.
+    -  On Host1, confd spots the new BGP address for hostB and, writes out a
+       configuration file for Bird.  This file contains the configuration 
+       necessary for a full mesh between all of the BGP peers (in this case
+       it is just peering with Host2).  Once the configuration if updated,
+       confd signals to BIRD to reload its configuration.
+    -  Similar happens on Host2.
+-  BIRD has been restarted on Host1 and Host2 and, with the updated 
+   configuration, has established BGP peering between the two hosts.  At the
+   moment there are no interesting routes programmed on either host and so
+   BIRD is largely idle.  BIRD has used the default global AS number
+   which was initialized by calicoctl to set up the peering.
+
+![calicoctl node](images/lifecycle/calicoctl_node.png)
+
+## 2. Create containers on Host
+
+User creates a Docker container on each host using the `docker run`  command:
+
+    # Host1
+    docker run --net=none --name=workload_A -tid ubuntu
+    
+    # Host2
+    docker run --net=none --name=workload_B -tid ubuntu
+
+### System response
+
+Two ubuntu workloads have been created, one on each host.  The `--net=none` 
+flag means that the containers are not networked through Docker bridged 
+networking, so the containers are completely independent and cannot access or 
+be accessed by outside sources.
+
+There are no changes to any component within the calico-node service.
+
+![docker run](images/lifecycle/docker_run.png)
+
+## 3. Add the containers to Calico Networking
+
+User adds the containers to Calico networking by using the `calicoctl container add`
+command.  An IP address for both containers is selected from the pre-configured
+IPv4 IP Pool.
+
+    # Host1
+    calicoctl container add workload_A 192.168.1.1
+    
+    # Host2
+    calicoctl container add workload_B 192.168.1.2
+
+### System Response
+
+The calicoctl utility:
+-  Checks the IP Pools configured in etcd to confirm that the container IP 
+   value supplied is within a configured pool 
+-  Creates a veth pair on the host that will be used for the new endpoint (the
+   container interface).  It leaves one end in the host's network namespace and
+   moves the other end into the container namespace and renames the interface
+   in the container
+-  Configures the IP address of the interface in the container
+-  Sets up a default route over the interface in the container via the host IP
+   (the same IP used for the BGP client)
+-  Configures the endpoint information in etcd (see below for etcd data format) 
+
+Felix monitors the endpoint information for its own host.  At the moment, no
+rules are configured to allow packets between the containers, so Felix does 
```

Comment:
This is wrong!!!

Felix does do something. It creates the route, which bird then picks up and distributes to the other bird.